Move inplace_is_vmap_compatible to BatchedTensorImpl.h

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49120 Add batched grad testing to gradcheck, turn it on in test_autograd
* #49119 Update accumulate_grad to support vmap
* **#49118 Move inplace_is_vmap_compatible to BatchedTensorImpl.h**

I need this in the next stack up. It seems useful to have as a helper
function.

Test Plan:
- run tests

Differential Revision: [D25563546](https://our.internmc.facebook.com/intern/diff/D25563546)

zou3519 committed:
Move inplace_is_vmap_compatible to BatchedTensorImpl.h

I need this in the next stack up. It seems useful to have as a helper
function.

Test Plan:
- run tests

[ghstack-poisoned]

zou3519 committed:
Update on "Move inplace_is_vmap_compatible to BatchedTensorImpl.h"


I need this in the next stack up. It seems useful to have as a helper
function.

Test Plan:
- run tests

[ghstack-poisoned]

zou3519 committed:
Update on "Move inplace_is_vmap_compatible to BatchedTensorImpl.h"


I need this in the next stack up. It seems useful to have as a helper
function.

Test Plan:
- run tests

[ghstack-poisoned]

zou3519 committed:
Update on "Move inplace_is_vmap_compatible to BatchedTensorImpl.h"


I need this in the next stack up. It seems useful to have as a helper
function.

Test Plan:
- run tests

[ghstack-poisoned]

zou3519 committed:
Update on "Move inplace_is_vmap_compatible to BatchedTensorImpl.h"


I need this in the next stack up. It seems useful to have as a helper
function.

Test Plan:
- run tests

[ghstack-poisoned]

ezyang reviewed (approved):
the camel case looks so weird

zou3519 committed:
Update on "Move inplace_is_vmap_compatible to BatchedTensorImpl.h"


I need this in the next stack up. It seems useful to have as a helper
function.

Test Plan:
- run tests

[ghstack-poisoned]
